snuba/util.py

```diff
@@ -275,3 +276,16 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
         return cast(F, wrapper)
 
     return decorator
+
+
+@dataclass
+class MergeInfo:
+    result_part_name: str
+    elapsed: float
+    progress: float
+    size: int
+
+    @property
+    # estimated time remaining in seconds
+    def estimated_time(self) -> float:
+        return self.elapsed / (self.progress + 0.0001)
```

Comment:
I would suggest not adding this to the util module. 
Besides the fact that there should be no util module as it become the kitchen sink, I think this belongs to where we keep the optimize code. Now, unfortunately, that is a module in the root directory.  I think you should create a new directory for the optimize work.  
Put it inside this https://github.com/getsentry/snuba/tree/master/snuba/clickhouse